### PR TITLE
Bump PF, React, Redux versions

### LINKED ISSUE #718: [Deps]: Bump PF version and support React 18
### Feature description

"@patternfly/patternfly": "4.217.1" -> "4.219.2",
"@patternfly/react-catalog-view-extension": "4.92.26" -> "4.92.55",
"@patternfly/react-core": "4.250.1" -> "4.258.3",
"@patternfly/react-icons": "4.92.6" -> "4.92.10",
"@patternfly/react-styles": "4.91.6" -> "4.91.10",
"@patternfly/react-table": "4.111.4" -> "4.111.33",
"@patternfly/react-virtualized-extension": "^4.71.1" -> "4.88.55",

Although React 18 is still a beta version in PF, we can still upgrade our product to React 18 since it's been tested a lot on PF, the official support for React 18 in PF will be next June as they said for PF5.

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Closes #718 

## Description

1. Bump PF version to the latest
2. Bump React to react 18 so we need to explicitly use `children`
3. Bump `react-redux` and `redux` version because it needs to be compatible with React 18 to work functionally

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: frontend/src/SDKInitialize.tsx [hunk 1]
```diff
@@ -24,7 +24,11 @@ const config: React.ComponentProps<typeof AppInitSDK>['configurations'] = {
   wsAppSettings: () => Promise.resolve({ host: '', subProtocols: [] }),
 };
 
-const SDKInitialize: React.FC = ({ children }) => {
+type SDKInitializeProps = {
+  children?: React.ReactNode;
```
Comment (andrewballantyne) [inline frontend/src/SDKInitialize.tsx:28]: I know the original type was optional -- are we sure we don't want to just go all in and make them required when needed? (this applies to all uses of `children` prop)

File: frontend/src/app/Header.tsx [hunk 2]
```diff
@@ -19,8 +20,7 @@ const Header: React.FC<HeaderProps> = ({ onNotificationsClick }) => {
           alt={`${ODH_PRODUCT_NAME} Logo`}
         />
       }
-      logoComponent={Link}
-      logoProps={{ to: '/' }}
+      logoProps={{ onClick: () => navigate('/') }}
```
Comment (andrewballantyne) [inline frontend/src/app/Header.tsx:23]: Why this change?

Comment (DaoDaoNoCode) [reply frontend/src/app/Header.tsx:23]: This is because there is a type error when we try to pass Link as the logoComponent
```
Type 'ForwardRefExoticComponent<LinkProps & RefAttributes<HTMLAnchorElement>>' is not assignable to type 'ReactNode'.
```

File: frontend/package.json [hunk 3]
```diff
@@ -54,24 +55,25 @@
...
-    "react": "^17.0.2",
+    "react": "^18.2.0",
     "react-cool-dimensions": "^2.0.5",
-    "react-dom": "^17.0.2",
-    "react-redux": "^7.2.1",
+    "react-dom": "^18.2.0",
+    "react-redux": "^8.0.4",
     "react-router": "^6.4.1",
     "react-router-dom": "^6.4.1",
-    "redux": "^4.0.5",
```
Comment (andrewballantyne) [inline frontend/package.json:63]: Did you remove redux or am I blind?

File: frontend/src/redux/actions/actions.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import axios from 'axios';
 import { ThunkAction } from 'redux-thunk';
 import { Actions, AppNotification, AppState, GetUserAction } from '../types';
-import { Action } from 'redux';
+import { Action } from '@reduxjs/toolkit';
```
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply frontend/src/redux/actions/actions.ts:4]: > Remove redux package and use @reduxjs/toolkit to totally replace it, also, follow the redux doc to create 2 hooks to better use dispatch and selector without type errors

Why?

File: frontend/package.json (hunk 3 above)
Comment (DaoDaoNoCode) [reply frontend/package.json:63]: yeah i removed it. https://redux.js.org/redux-toolkit/overview#purpose. I mentioned this in the description. It's lightweight and contains everything we need already. It's the same and officially maintained.

Comment (andrewballantyne) [reply frontend/package.json:63]: I don't understand... our goal is to get rid of redux in the long run -- why would you change our backing library?

Comment (DaoDaoNoCode) [reply frontend/package.json:63]: I can bring it back if you want, actually it makes no difference, we are not heavily using redux in our application so we only used a very small part of functions in it.

**Review by andrewballantyne — APPROVED** [review]
